jvm/src/main/scala/fr/hmil/roshttp/SSLUtils.scala

```diff
@@ -0,0 +1,98 @@
+package fr.hmil.roshttp
+
+import java.io._
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths}
+import java.security.cert.{CertificateFactory, X509Certificate}
+import java.security.{KeyStore, SecureRandom}
+import java.util.Objects.nonNull
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+import java.util.{Base64, Objects}
+
+import javax.net.ssl._
+
+object SSLUtils {
+
+  private val SSLCONTEXT_TYPE = "TLSv1.2"
+  private val KEYSTORE_TYPE = "PKCS12"
+  private val CERTIFICATE_TYPE = "X.509"
+  private val CERTIFICATE_HEADER = "-----BEGIN CERTIFICATE-----"
+  private val CERTIFICATE_FOOTER = "-----END CERTIFICATE-----"
+  private val CERTIFICATE_PATTERN = Pattern.compile(CERTIFICATE_HEADER + "(.*?)" + CERTIFICATE_FOOTER, Pattern.DOTALL)
+
+  private val NEW_LINE = "\n"
+  private val EMPTY = "";
+
+  def createSslContext(sslConfig: SSLConfig): SSLContext = {
+    val sslContext: SSLContext = SSLContext.getInstance(SSLCONTEXT_TYPE)
+    var keyManagers: Array[KeyManager] = null
+    var trustManagers: Array[TrustManager] = null
+
+    if (sslConfig.keyStorePath.isDefined && sslConfig.keyStorePassphrase.isDefined) {
+      keyManagers = createKeyManagers(sslConfig.keyStorePath.get, sslConfig.keyStorePassphrase.get)
+    }
+
+    if (sslConfig.trustStorePaths.isDefined) {
+      trustManagers = createTrustManagers(sslConfig.trustStorePaths.get)
+    }
+
+    sslContext.init(keyManagers, trustManagers, new SecureRandom())
```

Comment:
The doc says the last argument can be left empty, in which case the default implementation is used. Why not leave it empty?

Reply:
I wasn't aware that by passing null it would use the default implementation. I will refactor it